amd/src/courseformat/tabtree.js

```diff
@@ -0,0 +1,176 @@
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+import {BaseComponent} from 'core/reactive';
+import {getCurrentCourseEditor} from 'core_courseformat/courseeditor';
+import Templates from 'core/templates';
+
+
+/**
+ * Course section tabs updater.
+ *
+ * @module     format_multitopic/courseformat/tabtree
+ * @class      format_multitopic/courseformat/tabtree
+ * @copyright  2022 Jeremy FitzPatrick and Te Wānanga o Aotearoa
+ * @copyright  based on work by 2021 Ferran Recio <ferran@moodle.com>
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+export default class Component extends BaseComponent {
+
+    /**
+     * Constructor hook.
+     */
+    create() {
+        // Optional component name for debugging.
+        this.name = 'coursetabs';
+        // Default query selectors.
+        this.selectors = {
+            TAB: `ul:first-of-type li`,
+            CHILDTAB: `ul:nth-child(2) li`
+        };
+        // Default classes
+        this.classes = {
+            ACTIVETAB: 'active'
+        };
+        // Objects to keep tabs on the tabs
+        this.tabs = {};
+        this.childtabs = {};
+        this.activetab = 0;
+    }
+
+    getWatchers() {
+        return [
+            // Sections sorting.
+            {watch: `course.sectionlist:updated`, handler: this._refreshCourseSectionTabs},
+        ];
+    }
+
+    static init(target) {
+        return new Component({
+            element: document.getElementById(target),
+            reactive: getCurrentCourseEditor()
+        });
+    }
+
+    /**
+     * Initial state ready method.
+     *
+     */
+    stateReady() {
+        // Get tab elements.
+        const tabs = this.getElements(this.selectors.TAB);
+        for (let i = 0; i < tabs.length - 1; i++) { // Don't count last "add section" tab.
+            let tab = tabs.item(i);
+            let id = tab.querySelector("a div").dataset.itemid;
+            let classes = tab.querySelector("a").classList.value;
+            this.tabs[id] = tab;
+            if (classes.indexOf(this.classes.ACTIVETAB) !== -1) {
+                this.activetab = i;
+            }
+        }
+
+        const childtabs = this.getElements(this.selectors.CHILDTAB);
+        for (let i = 0; i < childtabs.length - 1; i++) { // Don't count last "add section" tab.
+            let tab = childtabs.item(i);
+            let id = tab.querySelector("a div").dataset.itemid;
+            this.childtabs[id] = tab;
+        }
+    }
+
+    /**
+     * Refresh the section tabs.
+     *
+     * @param {object} param
+     * @param {Object} param.element
+     */
+    _refreshCourseSectionTabs({element}) {
+        // Do things that make the first row tabs match firstsectionlist.
+        const toptabslist = element.firstsectionlist ?? [];
```

Comment:
Need to find second-level tab list earlier.
```suggestion
        const toptabslist = element.firstsectionlist ?? [];
        const childtabslist = element.secondsectionlist ?? [];
```